### ISSUE TITLE
supplying filterwarnings overrides default filters

### ISSUE THREAD

**Issue by jaraco** [pytest-dev/pytest#4013]
I'm really happy about pytest 3.8 and the work in #2908.

And now that I'm seeing warnings again, I realize how blissful ignorance is.

So I've gone through one project and fixed all of the deprecation warnings for the project under test (there are still many coming from other projects). In order to assert that the project no longer has any warnings, I added this to pytest.ini:

```
filterwarnings = 
    error:::mylib[.*]
```

Now tests pass, but all of the deprecation warnings (from the dependencies) are suppressed. That wasn't what I'd intended by setting the project to error on warnings.

I don't think that's the right behavior. Instead, pytest should configure the filters for DeprecationWarnings and PendingDeprecationWarnings and then allow the user to _override_ those filters but not suppress the defaults with any value.

If the project must suppress the defaults, the docs should at least describe what the defaults are so they can be copied into the configuration wishing to define other filters. I had to traverse to the source to determine what the defaults are:

```
filterwarnings =
	always::DeprecationWarning
	always::PendingDeprecationWarning
```

Is there any reason not to always configure the defaults and allow overrides?

**Comment by nicoddemus** [comment]
Hi @jaraco,

> Now tests pass, but all of the deprecation warnings (from the dependencies) are suppressed. That wasn't what I'd intended by setting the project to error on warnings.

Indeed. When we introduced the warnings plugin a few versions back we had some serious regressions, so this time around I decided to be really conservative to minimize disruption.

The unittest module only [changes the warnings filters to diplay deprecations](https://github.com/python/cpython/blob/db1a80e97aa7217c561fb3627f70be1882de9534/Lib/unittest/runner.py#L167) if [sys.warnoptions](https://github.com/python/cpython/blob/db1a80e97aa7217c561fb3627f70be1882de9534/Lib/unittest/main.py#L88) is set (usually by `PYTHONWARNINGS`). I decided to go with the same route with our warnings plugin:

https://github.com/pytest-dev/pytest/blob/58aa4f91f5e124921eba24baef6c62db5d4eb279/src/_pytest/warnings.py#L72-L90

So basically we only install the filters *if* the user did not specify any filter configuration themselves, either using pytest's mechanisms or python's own `PYTHONWARNINGS`.

But I agree with you that this is not ideal, because as you point out any customization effectively silences deprecation warnings again.

I believe that for `3.9` we can install the deprecation filters before all other filters, right after `catch_warnings()` in the previous snippet. This shows deprecations by default, but any other filter that comes after it will take precedence.
